Web Import: Add New Entries to 'Imported Entries' Group

paudelritij opened the pull request:
<!-- YOU HAVE TO MODIFY THE TEXT BELOW TO FIT YOUR PR. OTHERWISE, YOUR PR WILL BE CLOSED WITHOUT FURTHER COMMENT. -->

<!-- LINK THE ISSUE WITH THE "Closes" KEYWORD -->
<!-- Example: Closes (link) OR Closes #12345 -->

Closes #12548
This PR introduce feature to add new group "Imported Entries"

### To Do
- [ ] Link preferences configuration to BibEntry (for enable/disable and name of group)
- [ ] Configure Meta data for imported entries group 
- [ ] Introduce Smart Group `1 SmartGroup:Import entries\`
- [?] Create library-specific settings for particular imported entries group for name and disable/enable option
- [ ] Add test cases

### Done 
- [x] Add configurable field  for `Imported entries` group in preferences
- [x] Create fields for `Import entries` group in library preferences 

### Mandatory checks

<!--
Go through the checklist below. It is mandatory, even for a draft pull request.

Keep ALL the items. Replace the dots inside [.] and mark them as follows: 
[x] done 
[ ] not done 
[/] not applicable
-->

- [x] I own the copyright of the code submitted and I license it under the [MIT license](https://github.com/JabRef/jabref/blob/main/LICENSE)
- [.] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if change is visible to the user)
- [.] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [.] Screenshots added in PR description (if change is visible to the user)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

jabref-machine commented:
JUnit tests are failing. You can see which checks are failing by locating the box "Some checks were not successful" on the pull request page. To see the test output, locate "Tests / Unit tests (pull_request)" and click on it.

You can then run these tests in IntelliJ to reproduce the failing tests locally. We offer a quick test running howto in the section [Final build system checks](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-12-build.html#final-build-system-checks) in our setup guide.


<!-- ghprcomment
Tests
Unit tests
-->

jabref-machine commented:
Your code currently does not meet [JabRef's code guidelines](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). We use [Checkstyle](https://checkstyle.sourceforge.io/) to identify issues. You can see which checks are failing by locating the box "Some checks were not successful" on the pull request page. To see the test output, locate "Tests / Checkstyle (pull_request)" and click on it.

In case of issues with the import order, double check that you [activated Auto Import](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#enable-proper-import-cleanup). You can trigger fixing imports by pressing <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>O</kbd> to trigger [Optimize Imports](https://www.jetbrains.com/guide/tips/optimize-imports/).

Please carefully follow [the setup guide for the codestyle](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). Afterwards, please [run checkstyle locally](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#run-checkstyle) and fix the issues, commit, and push.


<!-- ghprcomment
Tests
Checkstyle
-->

jabref-machine commented:
Note that your PR will not be reviewed/accepted until you have gone through the mandatory checks in the description and marked each of them them exactly in the format of `[x]` (done), `[ ]` (not done yet) or `[/]` (not applicable).


<!-- ghprcomment
Tests
PR checklist OK
-->

jabref-machine commented:
Note that your PR will not be reviewed/accepted until you have gone through the mandatory checks in the description and marked each of them them exactly in the format of `[x]` (done), `[ ]` (not done yet) or `[/]` (not applicable).


<!-- ghprcomment
Tests
PR checklist OK
-->

jabref-machine commented:
Your code currently does not meet [JabRef's code guidelines](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). We use [Checkstyle](https://checkstyle.sourceforge.io/) to identify issues. You can see which checks are failing by locating the box "Some checks were not successful" on the pull request page. To see the test output, locate "Tests / Checkstyle (pull_request)" and click on it.

In case of issues with the import order, double check that you [activated Auto Import](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#enable-proper-import-cleanup). You can trigger fixing imports by pressing <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>O</kbd> to trigger [Optimize Imports](https://www.jetbrains.com/guide/tips/optimize-imports/).

Please carefully follow [the setup guide for the codestyle](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). Afterwards, please [run checkstyle locally](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#run-checkstyle) and fix the issues, commit, and push.


<!-- ghprcomment
Tests
Checkstyle
-->

jabref-machine commented:
Your code currently does not meet [JabRef's code guidelines](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). We use [Checkstyle](https://checkstyle.sourceforge.io/) to identify issues. You can see which checks are failing by locating the box "Some checks were not successful" on the pull request page. To see the test output, locate "Tests / Checkstyle (pull_request)" and click on it.

In case of issues with the import order, double check that you [activated Auto Import](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#enable-proper-import-cleanup). You can trigger fixing imports by pressing <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>O</kbd> to trigger [Optimize Imports](https://www.jetbrains.com/guide/tips/optimize-imports/).

Please carefully follow [the setup guide for the codestyle](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html). Afterwards, please [run checkstyle locally](https://devdocs.jabref.org/getting-into-the-code/guidelines-for-setting-up-a-local-workspace/intellij-13-code-style.html#run-checkstyle) and fix the issues, commit, and push.


<!-- ghprcomment
Tests
Checkstyle
-->

Siedlerchr commented:
@paudelritij  I need to do a follow up fix in the equals method because it was triggering a "library has changed" after writing because of fail in equality chek

https://github.com/JabRef/jabref/pull/13193/files#diff-a2132ee5e8b4f2170d651de9988fcf225240d8ba78ac9809eb6f58cd196fe58f

koppor commented:
> @paudelritij I need to do a follow up fix in the equals method because it was triggering a "library has changed" after writing because of fail in equality chek
> 
> https://github.com/JabRef/jabref/pull/13193/files#diff-a2132ee5e8b4f2170d651de9988fcf225240d8ba78ac9809eb6f58cd196fe58f

Could this have been determined using test cases?

paudelritij commented:
Thanks for let me know

> @paudelritij I need to do a follow up fix in the equals method because it was triggering a "library has changed" after writing because of fail in equality chek
> 
> [#13193 (files)](https://github.com/JabRef/jabref/pull/13193/files#diff-a2132ee5e8b4f2170d651de9988fcf225240d8ba78ac9809eb6f58cd196fe58f)

I agree that the equal method was not correct. Upon reviewing it now, I believe this should be what you did, and it should not use instanceof with an explicit group, which I had implemented. Regarding the 'library has changed' trigger, it was initially present while I was developing the smart group. I thought I had fixed it in the middle of adding the smart group to the codebase, but I am wondering how it was produced!



> > @paudelritij I need to do a follow up fix in the equals method because it was triggering a "library has changed" after writing because of fail in equality chek
> > [#13193 (files)](https://github.com/JabRef/jabref/pull/13193/files#diff-a2132ee5e8b4f2170d651de9988fcf225240d8ba78ac9809eb6f58cd196fe58f)
> 
> Could this have been determined using test cases?

I am keen to know more about this
